tests/unit/events/map_to_aois_behaviour_test.py

```diff
@@ -336,3 +336,64 @@ def test_events_map_to_aois_no_new_columns_when_all_aoi_columns_present(
 
     # Also ensure that pre-existing AOI columns remain untouched (no overwrite during concat).
     assert ev.frame.select('label').item() == 'pre'
+
+
+@pytest.mark.parametrize('preserve_structure', [True, False])
+def test_previous_saccades_untouched_after_map_to_aois(
+    simple_stimulus: TextStimulus,
+    preserve_structure: bool,
+) -> None:
+    """Events with preceding saccades remain unchanged by map_to_aois.
+
+    - AOIs are only mapped for fixation rows.
+    - Saccade rows keep their original fields (name/onset/offset and location or its components),
+      and receive only None values in the appended AOI columns.
+    """
+    # Build a frame where a saccade precedes a fixation (and another saccade follows).
+    base = pl.DataFrame(
+        {
+            'name': ['saccade', 'fixation', 'saccade'],
+            'onset': [0, 1, 2],
+            'offset': [1, 2, 3],
+            'location': [[5.0, 5.0], [5.0, 5.0], [15.0, 5.0]],
+        },
+    )
+
+    original = base.clone()
+
+    events = Events(data=base)
+    events.map_to_aois(simple_stimulus, preserve_structure=preserve_structure)
+
+    # only middle fixation is inside AOI 'A'
+    labels = events.frame.get_column('label').to_list()
+    assert labels == [None, 'A', None]
+
+    # Verify saccade rows are unchanged (except for expected structural handling)
+    if preserve_structure:
```

Comment:
please avoid if-else in tests. usually that's a sign that the function is two tests in one, so it should be split (or further parametrized).

the only thing that these two tests share is the test setup (repeated test setup/teardown code  is usually best implemented as a fixture), so it's probably best to just split the function.